src/transformers/modeling_utils.py

```diff
@@ -2005,40 +2105,56 @@ def _fix_key(key):
                             (checkpoint_key, state_dict[checkpoint_key].shape, model_state_dict[model_key].shape)
                         )
                         del state_dict[checkpoint_key]
+            return mismatched_keys
+
+        if low_cpu_mem_usage:
+            # XXX: move together with state_dict = None?
+            model_state_dict = None  # state_dict ref count to 0
+            move_model_to_meta(model, loaded_keys, start_prefix)
 
+        if state_dict is not None:
+            # Whole checkpoint
+            mismatched_keys = _find_mismatched_keys(
```

Comment:
Note, that I refactored the original code duplication - as it was a rather large complicated code that was identical. 

I had to parameterize the function rather than using a closure, since state_dict gets manifested in the case of low_mem/sharded after the function was defined.